### PR TITLE
Authentication v1

### PR DESCRIPTION
**Pull request by SonaliChavan95**
Only user login feature is included in the PR. 
I still need to work on signup form and validations, will create a new PR for it. 
This PR has seed data for user login using which we can authenticate for now. 

Trello: https://trello.com/c/D8pNo1iv/15-user-authentication

### CODE REVIEW

File: public/js/location.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
+(function () {
```
Comment (SonaliChavan95) [inline public/js/location.js:1]: I have used 3rd party API for fetching states and cities for US. 
If anyone is facing an issue on their local then let me know.

File: data/users.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const users = require('../seed');
+
+let exportedMethods = {
+  getUserByEmail(email) {
+    if (!email) throw 'You must provide a email';
+    if (typeof email != 'string' || email.trim().length === 0)
+      throw 'You must provide a valid email';
+
+    const user = users.filter((user) => user.email === email);
```
Comment (abhinav098) [inline data/users.js]: You can use `users.find` here to avoid using user[0].

File: routes/users.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+router.get('/logout', async (req, res) => {
+  req.session.destroy();
+  return res.redirect('/');
+  // res.render('customError', {
+  //   title: 'Logout',
+  //   message: 'You have been succesfully logged out of the application.',
+  // });
+});
+
+router.get('/new', async (req, res) => {
```
Comment (abhinav098) [inline routes/users.js:56]: @SonaliChavan95 Could you please rename this route to `/signup`?

File: routes/index.js [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
-  app.use('/login', loginRoutes);
+  app.use('/users', userRoutes);
   app.use('/donations', donationRoutes);
 
   // static paths
   app.get('/', async (req, res) => {
-    res.status(200).render('static/home', { title: 'Home' });
+    res.status(200).render('static/home', {
+      title: 'Home',
+      authenticated: req.session.user ? true : false,
```
Comment (abhinav098) [inline routes/index.js:12]: 👍

**Review by abhinav098 — CHANGES_REQUESTED** [review]
Looks good. Added two comments.

**Review by abhinav098 — DISMISSED** [review]
Approved. Awesome Work !!

**Pull request merged**